### PR TITLE
Cleanup date related template tags

### PR DESCRIPTION
**Pull request by rowanseymour**
* Replace `format_datetime` with `|datetime`.. except where we use it in emails
 * Replace `short_datetime` with new `|timedate` filter
 * Remove unused template filters
 * Move all tests for those tags/filters into `temba.utils.templatetags.tests`

### CODE REVIEW

File: temba/utils/templatetags/temba.py [hunk 1]
```diff
@@ -153,17 +153,23 @@ def to_json(value):
 
 @register.filter
 def duration(date):
-    return mark_safe(f"<temba-date value='{date.isoformat()}' display='duration'></temba-date>")
+    return mark_safe(
+        f'<temba-date value="{date if isinstance(date, str) else date.isoformat()}" display="duration"></temba-date>'
```
Comment (rowanseymour) [inline temba/utils/templatetags/temba.py:157]: there are places where we have the date already as an ISO string

File: templates/channels/channellog_call.html [hunk 2]
```diff
@@ -17,7 +17,7 @@
   </div>
   <div class="inline-flex mr-8">
     <div class="font-normal mr-2">{% trans "Date" %}</div>
-    {% format_datetime call.created_on seconds=True %}
```
Comment (rowanseymour) [inline templates/channels/channellog_call.html:20]: now that you can hover over these things to see the full timestamp, less import to show seconds everywhere

File: temba/utils/templatetags/temba.py [hunk 3]
```diff
@@ -225,16 +229,6 @@ def format_datetime(context, dt, seconds: bool = False):
     return datetime_to_str(dt, fmt, tz)
 
 
-@register.filter
-def parse_isodate(value):
-    return iso8601.parse_date(value)
-
-
-@register.filter
-def first_word(value):
```
Comment (rowanseymour) [inline temba/utils/templatetags/temba.py:234]: not used

File: temba/utils/templatetags/temba.py [hunk 4]
```diff
@@ -225,16 +229,6 @@ def format_datetime(context, dt, seconds: bool = False):
     return datetime_to_str(dt, fmt, tz)
 
 
-@register.filter
-def parse_isodate(value):
```
Comment (rowanseymour) [inline temba/utils/templatetags/temba.py:229]: no longer used

File: temba/utils/templatetags/temba.py [hunk 5]
```diff
@@ -102,34 +88,6 @@ def annotated_field(field, label, help_text):
     return field.as_widget(attrs=attrs)
 
 
-@register.filter("delta", is_safe=False)
-def delta_filter(delta):
```
Comment (rowanseymour) [inline temba/utils/templatetags/temba.py:106]: not used

File: templates/contacts/contactgroup_read.html [hunk 6]
```diff
@@ -1,105 +0,0 @@
-{% extends "smartmin/read.html" %}
```
Comment (rowanseymour) [inline templates/contacts/contactgroup_read.html:1]: this hasn't been a view for a very long time...

File: temba/utils/templatetags/temba.py [hunk 7]
```diff
@@ -62,20 +48,6 @@ def oxford(forloop, conjunction=_("and")):
     return ""
 
 
-@register.filter
-def icon(o):
```
Comment (rowanseymour) [inline temba/utils/templatetags/temba.py:66]: not used

**Review by norkans7 — APPROVED** [review]

**Pull request merged**